### DISCUSSION TITLE
Map display issues in China deployment

### DISCUSSION THREAD

**Discussion by wdlier** [wavelog/wavelog#714] [Q&A] [answered]
Hello, may I ask if modifying the map data should be done here? I have modified the data but it still doesn't display.
![1](https://github.com/user-attachments/assets/8664acf5-b968-4816-b708-eb74840fedf4)
![2](https://github.com/user-attachments/assets/1b61b547-19e5-4080-beb2-5c86cbbb17b4)
![3](https://github.com/user-attachments/assets/7aad1e0d-a297-4744-9f2f-154c0dc7ab6c)

**Comment by HB9HIL** [comment]
Hello @wdlier 

maybe @violarulan can help here?

**Comment by violarulan** [comment]
Tested with Tianditu and it works fine.

![Snipaste_2024-08-08_09-17-52](https://github.com/user-attachments/assets/67e8445c-23e5-41e6-85ee-31bfb39f94d9)

![Snipaste_2024-08-08_09-32-08](https://github.com/user-attachments/assets/271c31a3-4e8e-4913-95f4-1496f95819e6)

I've seen there are other errors in your console output. Can you check at these first? It seems that `base_url` is not configured.

**Comment by HB9HIL** [comment]
@wdlier 

Which Wavelog Version do you use and how is it installed?

**Reply by wdlier** [reply]
Version | 1.7

Thank you for your reply. I have contacted @violarulan.

**Comment by wdlier** [comment]
![image](https://github.com/user-attachments/assets/739f0869-87ac-4978-b14d-df27ba109cf6)
Thank you very much, I have successfully made the changes.

操作如下：
![image](https://github.com/user-attachments/assets/c2a65e96-9273-4c6f-a474-b863f901678b)

https://webrd0{s}.is.autonavi.com/appmaptile?lang=zh_cn&size=1&scale=1&style=7&x={x}&y={y}&z={z}

![image](https://github.com/user-attachments/assets/bc0c2417-dd5d-4c9b-82bc-12795110ee26)

修改完，重启apache服务。
sudo systemctl restart apache2（基于ubuntu）

清除浏览器缓存，登入即可修改成功

**Comment by HB9HIL** [comment]
Great to see the community helps esch other :) 

@violarulan Can we mark something here as answer so future users with the same issue know what to do?

**Comment by violarulan** [comment, marked as answer]
Sure thing o7. @HB9HIL 

Steps to change map provider:

1. Change `map_tile_server` in table `options` in database to the new provider. See [Leaflet Docomentation](https://leafletjs.com/reference.html#tilelayer) for URL template info.
2. **Optional:** If your provider doesn't use `abc` subdomains (OSM does use it), you have to change the subdomain parameter. Replace `abc` with `<new subdomain identifier>` in `/assets/js/leaflet/leaflet.js`. 

For example, AutoNavi tile services uses such domains:

```
https://webrd01.is.autonavi.com
https://webrd02.is.autonavi.com
...
https://webrd04.is.autonavi.com
```

The `map_tile_server` row will be `https://webrd0{s}.is.autonavi.com/appmaptile?lang=zh_cn&size=1&scale=1&style=7&x={x}&y={y}&z={z}`
Then replace `abc` with `1234` in `/assets/js/leaflet/leaflet.js`

All set.

---

The reason why we change the `leaflet.js` is that there are many places that call the Leaflet. Changing all of them will be a big task. The solution here is not very beautiful but quick.

**Comment by violarulan** [comment]
In version 1.9, [Static Map Image](https://github.com/wavelog/wavelog/wiki/Static-Map-Images) is introduced.👏

The image is generated by PHP. If a subdomain combination other than `abc` (OSM use this) is used, you should edit `application/models/Staticmap_model.php`.

Find all `new \Wavelog\StaticMapImage\TileLayer` constructors, append the new subdomain you would like to use to them, 1234` for example.

```
$tileLayer = new \Wavelog\StaticMapImage\TileLayer($server_url, $attribution, $thememode, '1234');
```
![Snipaste_2024-12-06_09-44-18](https://github.com/user-attachments/assets/08a12e5e-fe72-46c1-83c9-c692f77ff515)

**Reply by HB9HIL** [reply]
Probably better I create an override for that so you don't have to edit the code.
